tests/testthat/helper-funs.R

```diff
@@ -40,30 +40,105 @@ funList <-
 
 expectDopp2 <- function(funName, ggplot2Version) {
   path <- paste0("ggMarginal/ggplot2-", ggplot2Version)
-  vdiffr::expect_doppelganger(funName, funList[[funName]](), path = path)
+  vdiffr::expect_doppelganger(
+    funName, 
+    printMuffled(funList[[funName]]()),
+    path = path
+  )
 }
 
-# withGGplot2Version is essentially the same function as with_pkg_version that
+# withVersions is essentially the same function as with_pkg_version that
 # appears here: https://gist.github.com/jimhester/d7aeb95bbed02f2985a87c2a3ede19f5.
-# This function allows us to run unit tests under different versions of ggplot2.
-withGGplot2Version <- function(ggplot2Version, code) {
-  if (isNamespaceLoaded("ggplot2")) {
-    unloadNamespace("ggplot2")
-  }
+# This function allows us to run unit tests under different versions of ggplot2,
+# confirming that ggMarginal works under all versions >= 2.2.0. We also set
+# the package versions of three packages (vdiffr, fontquiver, and svglite) 
+# that could slightly effect the rendering of the SVGs, thus causing the tests 
+# to fail.
+withVersions <- function(..., code) {
+  
+  packageVersions <- list(...)
+  
+  unloadPackages(packages = names(packageVersions))
+  on.exit(unloadPackages(packages = names(packageVersions)))
+  
   dir <- tempfile()
   dir.create(dir)
   on.exit(unlink(dir))
-  withr::with_libpaths(dir, action = "prefix", {
-    on.exit(unloadNamespace("ggplot2"))
+  names(packageVersions)
```

Comment:
this line and the next don't seem to do anything?

Reply:
Yes, they should be removed.